src/core/lib/gprpp/memory.h

```diff
@@ -43,8 +43,29 @@ inline void Delete(T* p) {
 }
 
 // TODO(veblush): Remove this after removing all usages.
+class DefaultDeleteChar {
+ public:
+  void operator()(char* p) {
+    if (p == nullptr) return;
+    gpr_free(p);
+  }
+};
+
+// TODO(veblush): Remove this after removing all usages.
+template <typename T>
+struct ResolveDeleter {
+  using deleter = std::default_delete<T>;
+};
+template <>
+struct ResolveDeleter<char> {
+  using deleter = DefaultDeleteChar;
+};
+
+// TODO(veblush): Remove this after removing all usages.
+// This is equivalent to std::unique_ptr except that it uses gpr_free
+// for deleter only for UniquePtr<char>
```

Comment:
It seems very strange to make the behavior special for `char`.  If that's the only type we ever use `UniquePtr<>` for anymore, why not just use `gpr_free()` for everything here?

Reply:
Yes, this is pretty strange but this is because we have a few internal cases using UniquePtr for non-char yet. Once I remove all cases using `grpc_core::UniquePtr` for non-char type in internal sources, I will change this not to allow non-char type at all.